packages/editor/src/components/properties/MediaSourceProperties.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import InputGroup from '../inputs/InputGroup'
+import BooleanInput from '../inputs/BooleanInput'
+import { EditorComponentType, updateProperty } from './Util'
+import { useTranslation } from 'react-i18next'
+import NumericInput from '../inputs/NumericInput'
+import { getComponent } from '@xrengine/engine/src/ecs/functions/ComponentFunctions'
+import { MediaComponent } from '@xrengine/engine/src/scene/components/MediaComponent'
+
+export const MediaSourceProperties: EditorComponentType = (props) => {
+  const { t } = useTranslation()
+
+  const mediaComponent = getComponent(props.node.entity, MediaComponent)
+
+  return (
+    <>
+      <InputGroup
```

Comment:
we can populate views of such components from a JSON, since same HTML is being repeated again and again

Reply:
This is a good way to go. But currently all the editors for all nodes are implemented in this way and refactoring all of them is a big task in itself. Hence I am taking a note and will revisit this later.